LMSBorrower/src/main/java/com/lms/LMSBorrower/BorrowerDAO/BorrowerDAO.java

```diff
@@ -1,16 +1,170 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import com.lms.LMSBorrower.POJO.*;
+
+@Component
 public class BorrowerDAO {
 	
+	public Connection getConnection() throws SQLException{
+		  Connection con = null;
+		 try {
+			 con = (Connection) DriverManager.getConnection("jdbc:mysql://localhost:3306/library?useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC","root","TapHouse123");
+	         } 
+		 catch(SQLException e) {
+	            System.out.println(e);
+	       }
+		 return con;
+	    }
+	//get cardNo
+	public List<Borrower> ifCardExists() throws SQLException {
+		String sql = "SELECT * FROM tbl_borrower ";
+		PreparedStatement ps = getConnection().prepareStatement(sql);
+		ResultSet rs = ps.executeQuery();
+		
+		List<Borrower> brw = new ArrayList<Borrower>();
+		 
+		while (rs.next()) {
+			Borrower borrower = new Borrower();
+			borrower.setCardNo(rs.getInt("cardNo"));
+			borrower.setName(rs.getString("name"));
+			borrower.setAddress(rs.getString("address"));
+			borrower.setPhone(rs.getString("phone"));
+			brw.add(borrower);
+		}
+		return brw;
+	}
+	
 	
+	//show branch for checkout 
+	public List<LibraryBranch>  displayBranchCheckout() throws SQLException {
+		String sql = "SELECT * from tbl_library_branch";
+		PreparedStatement ps = getConnection().prepareStatement(sql);
+		ResultSet rs = ps.executeQuery();
+		
+		List<LibraryBranch> bch = new ArrayList<LibraryBranch>();
+		
+		while (rs.next()) {
+			 LibraryBranch branch = new LibraryBranch();
+			 branch.setBranchId(rs.getInt("branchId"));
+			 branch.setBranchName(rs.getString("branchName"));
+			 branch.setBranchAddress(rs.getString("branchAddress"));
+			 bch.add(branch);	
+			}
+		return bch;
+		}
 	
-	public Borrower getBorrowerByCardNo(int cardNo) {
-		Borrower bor = new Borrower();
-		 bor.setCardNo(1);
-		 bor.setName("Sayana");
-		 return bor;
-	}
+	//show  book for checkout
+	public List<Book>  displayBookCheckout(int branchId) throws SQLException {
+		String sql= ("SELECT tbl_book.bookId, CONCAT(tbl_book.title, ' by ' , tbl_author.authorName) AS title  FROM tbl_book INNER JOIN tbl_author ON tbl_book.authId = tbl_author.authorId INNER JOIN tbl_book_copies ON tbl_book.bookId =  tbl_book_copies.bookId INNER JOIN tbl_library_branch ON tbl_book_copies.branchId = tbl_library_branch.branchId WHERE tbl_library_branch.branchId = ? AND tbl_book_copies.noOfCopies >=1 ");
+		PreparedStatement ps = getConnection().prepareStatement(sql);
+		ps.setInt (1, branchId);
+		ResultSet rs = ps.executeQuery();
+		
+		List<Book> bk = new ArrayList<Book>();
+		
+		while (rs.next()) {
+			Book book = new Book();
+			book.setBookId(rs.getInt("bookId"));
+			book.setTitle(rs.getString("title"));
+			bk.add(book);	
+			}
+		return bk;
+		}
+	
+	//checkout finalize
+		public ResponseEntity <String> writeLoans( int cardNo, int branchId, int bookId) throws SQLException  {
```

Comment:
Your  DAO methods should be mainly void methods. All response entity returns should only be done in the controller.

Reply:
that has been taken care of in my next commit.